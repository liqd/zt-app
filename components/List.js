import React from 'react'
import { View } from 'react-native'
import { TouchableOpacity } from 'react-native'
import IconSLI from 'react-native-vector-icons/SimpleLineIcons'

import { styles } from './List.styles'
import { TextSourceSans } from './TextSourceSans'

export const ListContainer = (props) => {
  return (
    <View style={styles.listContainer}>
      <TextSourceSans style={styles.listTitle}>{props.title}</TextSourceSans>
      {props.children}
    </View>
  )
}

export const ListItem = (props) => {
  return (
    <View style={styles.listItem}>
      {props.children}
    </View>
  )
}

export const ListLink = (props) => {
  const arrowRightIcon = <IconSLI name='arrow-right' size={20}/>

  return (
    <View style={styles.listItem}>
      <TouchableOpacity
        accessibilityRole="button"
        onPress={props.onPress}
        style={styles.listLink}
        disabled={props.disabled}
      >
        <TextSourceSans style={styles.listLinkTitle} numberOfLines={1}>
          {props.linkTitle}
        </TextSourceSans>
        <TextSourceSans style={styles.listLinkTitle}>
          {arrowRightIcon}
        </TextSourceSans>
      </TouchableOpacity>
    </View>
  )
}
