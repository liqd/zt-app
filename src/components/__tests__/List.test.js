import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

import { ListLink } from '../List'

test('Test ListLink Snapshot', () => {
  const { toJSON } = render(<ListLink />)
  expect(toJSON()).toMatchSnapshot()
})

test('Test ListLink onPress', () => {
  const onPressMock = jest.fn()
  const { getByRole } = render(<ListLink onPress={onPressMock} disabled={false} />)
  const button = getByRole('button')
  fireEvent.press(button)
  expect(onPressMock).toHaveBeenCalled()
})

test('Test ListLink onPress disabled', () => {
  const onPressMock = jest.fn()
  const { getByRole } = render(<ListLink onPress={onPressMock} disabled={true} />)
  const button = getByRole('button')
  fireEvent.press(button)
  expect(onPressMock).not.toHaveBeenCalled()
})
