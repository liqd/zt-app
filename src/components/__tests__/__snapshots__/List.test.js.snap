// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ListLink Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderBottomColor": "#d8d8d8",
      "borderBottomWidth": 1,
      "color": "#1a1a1a",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "padding": 15,
      "width": "100%",
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontFamily": "SourceSansPro_400Regular",
          "fontSize": 36,
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "SourceSansPro_400Regular",
          "fontSize": 36,
        }
      }
    >
      <Text />
    </Text>
  </View>
</View>
`;
